### PR TITLE
GraphQl-439: Improve coverage for mutation addSimpleProductToCart or Cart contains a product after product got status Out of Stock

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
### Description (*)
Issue https://github.com/magento/graphql-ce/issues/439

#### Steps to reproduce (*)
Create Simple Product with
Name Simple Product
SKU: simple-product
Weight: 1
Price: 1
Qty: 1000
Create empty cart as a Guest or as Customer (GraphQL)
Add simple product to Cart (GraphQL)
Go to admin
Set Simple Product to be Out of Stock
Add Simple Product to Cart again (GraphQL) and get this error message

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddProductsToCart.php [hunk 1]
```diff
@@ -50,16 +50,16 @@ public function __construct(
      */
     public function execute(Quote $cart, array $cartItems): void
     {
-        foreach ($cartItems as $cartItemData) {
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Cart/AddProductsToCart.php:53]: We can't move this logic after error checking.

Now it works as:
1) Perform adding products to cart
2) Check errors after the previous operation

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddProductsToCart.php (hunk 1 above)
Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Cart/AddProductsToCart.php:53]: changes were reverted

**Comment by galaoleksandr** [comment]
Cannot reproduce. Looks like it is already fixed.

**Comment by lenaorobei** [comment]
@XxXgeoXxX could you please address build failures.

**Comment by lenaorobei** [comment]
Please instead of new mutation please introduce new field.

**Comment by lenaorobei** [comment]
Closing this PR since no updates for more than 2 weeks. Please feel free to reopen if you want to continue working on this task. Thank  you.

**Comment by TomashKhamlai** [comment]
Hello @XxXgeoXxX
Please merge the latest 2.3-develop. Schema is not loading on your PR. It can take a lot of time to find out why but merging locally fixed schema loading.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please use this approach https://github.com/magento/graphql-ce/pull/969/files#diff-0b2dca4fe975cc5dd9163cf736cd94daR134-R139

**Comment by TomashKhamlai** [comment]
This PR introduces some improvements:
:heavy_check_mark: I was not able to place the order
:x: `query cart` shows that the product is in the cart

**Comment by TomashKhamlai** [comment]
@XxXgeoXxX, please read this comment https://github.com/magento/graphql-ce/pull/475#pullrequestreview-306223781

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are planning to continue working on it using proposed approach.

**Pull request closed without merging**